samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js

```diff
@@ -1,45 +1,134 @@
+                    },
+                };
+            }
+            throw e;
+        }
+       
+    }
+    async getSingleMember(context) {
+        try {
+           const member = await TeamsInfo.getMember(
+                context,
+                context.activity.from.id
+            );
+            return member.name;
+        } catch (e) {
+            if (e.code === 'MemberNotFoundInConversation') {
+                context.sendActivity(MessageFactory.text('Member not found.'));
+                return e.code;
+            } 
+        throw e;        
+        }   
+    }
+}
+
+function GetJustInTimeCardAttachment() {
+    return CardFactory.adaptiveCard({
+        actions: [
+            {
+                type: 'Action.Submit',
+                title: 'Continue',
+                data: { msteams: { justInTimeInstall: true } }
+            },
+        ],
+        body: [
+            {
+                text: 'Looks like you have not used Action Messaging Extension app in this team/chat. Please click **Continue** to add this app.',
+                type: 'TextBlock',
+                wrap: 'bolder'
+            },
+        ],
+        type: 'AdaptiveCard',
+        version: '1.0',
+    });
+}
+
+function GetAdaptiveCardAttachment() {
+    return CardFactory.adaptiveCard({
+        actions: [{ type: 'Action.Submit', title: 'Close' }],
+        body: [
+            {
+                text: 'This app is installed in this conversation. You can now use it to do some great stuff!!!',
+                type: 'TextBlock',
+                isSubtle: false,
+                warp: true
+            },
+        ],
+        type: 'AdaptiveCard',
+        version: '1.0',
+    });
 }
 
 function createCardCommand(context, action) {
     // The user has chosen to create a card by choosing the 'Create Card' context menu command.
     const data = action.data;
     const heroCard = CardFactory.heroCard(data.title, data.text);
     heroCard.content.subtitle = data.subTitle;
-    const attachment = { contentType: heroCard.contentType, content: heroCard.content, preview: heroCard };
+    const attachment = {
+        contentType: heroCard.contentType,
+        content: heroCard.content,
+        preview: heroCard,
+    };
 
     return {
         composeExtension: {
             type: 'result',
             attachmentLayout: 'list',
-            attachments: [
-                attachment
-            ]
-        }
+            attachments: [attachment]
+        },
     };
 }
 
 function shareMessageCommand(context, action) {
     // The user has chosen to share a message by choosing the 'Share Message' context menu command.
     let userName = 'unknown';
-    if (action.messagePayload.from &&
-            action.messagePayload.from.user &&
-            action.messagePayload.from.user.displayName) {
+    if (
+        action.messagePayload.from &&
+        action.messagePayload.from.user &&
+        action.messagePayload.from.user.displayName
+    ) {
```

Comment:
Should be able to simplify to:

```suggestion
    if (action.messagePayload?.from?.user?.displayName) {
```